### PR TITLE
Various documentation fixes

### LINKED ISSUE #9322: Maximum height of nodebox collisionboxes is undocumented
##### Minetest version
```
5.1.0
```

##### Summary
Nodes are allowed to have an overhigh collisionbox (i.e. higher than 1).
However, this fact is not clearly documented in `lua_api.txt` yet. Especially the maximum permitted height is not documented.

##### Analysis

From experiments I know that currently (5.1.0), a height of exactly 2 (from y=-0.5 to y=1.5) does not function. Entities will behave very very broken when standing on this node.
However, a height slightly below 2 will work.

Maximum confirmed functional y value is 1.484375 = 95/64. This is the maximum height of leveled nodeboxes. I figured this out by experimenting with a leveled nodebox.

I conclude the maxium possible collisionbox Y must lie somewhere between 1.484375 and 1.5. I have not conducted further experiments so far.

### LINKED ISSUE #10105: minetest.get_us_time overflow
_Slackware Linux 14.2 x86_

get_us_time() will overflow and reset to zero every now and then. Had it happen several times in one day.
This should probably be documented, or ideally caught and worked around.

Update: This actually happened on a Linux server, not Windows. Oops.

### LINKED ISSUE #10470: Deprecated set_sky is not documented
##### Minetest version
Since 5.1.0

##### Summary
Problem 1: The old `set_sky` parameters are not documented
Problem 2: Compatibility in mods is not obvious -> presence check for `get_sky_color`?

### LINKED ISSUE #10682: CSM: core.localplayer is nil
##### Minetest version
```
5.3
```

##### OS / Hardware

Operating system: Windows
CPU: Core i7

##### Summary
core.localplayer is nil

### PR DESCRIPTION
**Pull request by SmallJoker**
Fixes #9322
Fixes #10105
Fixes #10682
Fixes #10470

## To do

This PR is Ready for Review.

### CODE REVIEW

**Comment by Wuzzy2** [comment]
> Extents greater than 1.45 into one direction may cause collision issues.

Does this mean the overall width/height of a node, or does this mean the distance from the center point, or does it mean something completely different?

**Comment by SmallJoker** [comment]
Distance from the center point into x/y/z directions. Would you have any suggestion for a better wording?

File: doc/lua_api.txt [hunk 1]
```diff
@@ -4347,6 +4348,8 @@ Utilities
           object_step_has_moveresult = true,
           -- Whether get_velocity() and add_velocity() can be used on players (5.4.0)
           direct_velocity_on_players = true,
+          -- New set_sky() syntax, including set_sun and set_moon functions
```
Comment (sfan5) [inline doc/lua_api.txt:4351]: since modders can test if set_sun is available this is not strictly needed, is it?

**Comment by Wuzzy2** [comment]
@sfan5: Because `set_sun` is not a new function. The text refers to the new *syntax* of `set_sun` (and `set_moon`).

**Comment by sfan5** [comment]
Wrong, `set_sun`, `set_moon` and `set_stars` *are* new functions.

File: doc/lua_api.txt (hunk 1 above)
Comment (SmallJoker) [reply doc/lua_api.txt:4351]: Right. I added this one to see whether a feature or a run-time check is preferred. If there are no objections from modders I'll revert this one and point out that the presence of `set_sun` or `set_moon` should be checked instead.

Comment (sfan5) [reply doc/lua_api.txt:4351]: Yeah that'd be my preferred approach.

**Comment by Wuzzy2** [comment]
But I found all 3 functions in `lua_api.txt` in 5.3.0. They work, too.

https://github.com/minetest/minetest/blob/5.3.0/doc/lua_api.txt

**Comment by SmallJoker** [comment]
> But I found all 3 functions in lua_api.txt in 5.3.0. They work, too.

But they don't work prior to 946c03c (5.3.0-dev). This new documentation helps modders to write backwards-compatible mods, and also to look up the behaviour of the previous notation.

Comments addressed. Please re-review.

**Review by sfan5 — APPROVED** [review]

**Comment by Wuzzy2** [comment]
>        --  ^ To avoid collision issues, keep each value
>        --    within the range of +/- 1.45.

Oh, OK. In that case, DevTest needs updating. `testnodes:nodebox_overhigh2` is a node that currently violates this rule, it goes to Y=1.50 atm. I was unaware of the allowed limit back when I made DevTest.

Please fix this (and don't forget the name/comment of the node).

Besides: Are you sure about the `1.45` number? Where does this number come from? Note that with `leveled` nodes, you can actually go higher than that, all the way to Y = 1.484375 = 1.5-1/64 by setting `param2=255`. That's a bit inconsistent here.

I have tested `testnodes:nodebox_leveled` with `param2=255` and it still worked. I can walk on it without problems, also dropped items behave correctly. But it is always possible I overlooked a bug.

**Comment by SmallJoker** [comment]
> That's a bit inconsistent here.

Yes sure you can, but that depends on how often the physics calculations run per second. But server-side physics are calculated less often, hence this rather limited value of `1.45` which should be safe enough for both sides.

**Comment by Wuzzy2** [comment]
Hmmmm, but if you suggest that using a nodebox with max Y > 1.45, even for `leveled`, shouldn't `lua_api.txt` then warn about this (practical) limit for `leveled` nodes as well then?

The highest "safe" value for `param2` would be `124` then (for `1.4375`), not `127` (for `1.484375`).

**Comment by SmallJoker** [comment]
Done

**Pull request merged**